feat(extension): Block on missing required config (#2187)
canonical/charmcraft: docs/reference/extensions/fastapi-framework-extension.rst

        token:
          description: The token for the service.
          type: string
          optional: false

In addition to this, you can set the configuration options to be
mandatory by setting the ``optional`` key to ``false``. This will
block the charm and stop services until the configuration is supplied. For example,
if your application needs an ``api-token`` to function correctly you can set
``optional``, as shown below. This will block the charm and stop the
services until the ``api-token`` configuration is supplied.

.. code-block:: yaml

> 

Reviewer: ```suggestion
    :caption: charmcraft.yaml
```

    config:
      options:
        api-token:
          description: The token necessary for the service to run.
          type: string
          optional: false

> .. note::
> 
>     A configuration with the ``optional: false`` option cannot have a
>      ``default`` value. If you set a ``default`` value to a configuration
>      with ``optional: false`` ``charmcraft`` will fail when packing the charm.

Reviewer: ```suggestion
.. note::

    A configuration with ``optional: false`` can't also have a ``default`` key.
    If it has both, the charm will fail to pack.
```
